scripts/scr_start_load/scr_start_load.gml

```diff
@@ -1,346 +1,228 @@
-	                    // if (obj_ini.mobi[com,ey]="Jump Pack") then company_size+=1;
-	                }
-	            }
-	            remove_size+=company_size;
-	            obj_ini.ship_carrying[shiyp]=company_size;
-	            obj_ini.man_size-=company_size;
-	            company_size=0;
-	        }
-	        if (squeeze=1){
-	            com+=1;ey=0;company_size=0;companies_loaded=2;
-	            repeat(500){ey+=1;
-	                if (obj_ini.role[com,ey]!=""){
-	                    obj_ini.lid[com,ey]=shiyp;
-	                    // obj_ini.uid[com,ey]=obj_ini.ship_uid[shiyp];
-	                    obj_ini.wid[com,ey]=0;obj_ini.loc[com,ey]=obj_ini.ship_location[shiyp];
-	                    company_size+=1;
-	                    if (obj_ini.armor[com,ey]="Terminator Armor") then company_size+=1;
-	                    if (obj_ini.armor[com,ey]="Tartaros") then company_size+=1;
-	                    if (obj_ini.armor[com,ey]="Dreadnought") then company_size+=7;
-	                    // if (obj_ini.mobi[com,ey]="Jump Pack") then company_size+=1;
-	                }
-	            }
-	            remove_size+=company_size;
-	            obj_ini.ship_carrying[shiyp]+=company_size;
-	            obj_ini.man_size-=company_size;
-	            company_size=0;
-	        }
-    
-	    }
-	}
-
-
-
-
-
-
-
-
-repeat (500){
-    ey += 1;
+    repeat (10) {
+        com += 1;
+        ey = 0;
+
+        repeat (500) {
+            ey += 1;
+
+            if (obj_ini.role[com, ey] != "") {
+                var n_size;
+                n_size = 1;
+
+                if (obj_ini.armor[com, ey] == "Terminator Armor") then n_size += 1;
+                if (obj_ini.armor[com, ey] == "Tartaros") then n_size += 1;
+                if (obj_ini.armor[com, ey] == "Dreadnought") then n_size += 5;
+                if (obj_ini.role[com, ey] == "Chapter Master") then n_size += 1;
+
+                if (obj_ini.ship_carrying[shiyp] + n_size > obj_ini.ship_capacity[shiyp]) {
+                    remove_size += company_size;
+                    obj_ini.ship_carrying[shiyp] += company_size;
+                    obj_ini.man_size -= company_size;
+                    company_size = 0;
+                    shiyp += 1;
+                    companies_loaded = 1;
+                }
+
+                obj_ini.lid[com, ey] = shiyp;
+                obj_ini.wid[com, ey] = 0;
+                obj_ini.loc[com, ey] = obj_ini.ship_location[shiyp];
+                company_size += 1;
+
+                if (obj_ini.armor[com, ey] == "Terminator Armor") then company_size += 1;
+                if (obj_ini.armor[com, ey] == "Tartaros") then company_size += 1;
+                if (obj_ini.armor[com, ey] == "Dreadnought") then company_size += 5;
+
+                // Load vehicles onto the ship
+                var v;
+                for (v = 1; v <= 5; v++) {
+                    if (obj_ini.veh_role[com, ey] == "Rhino") {

```

Comment:
would be nice if we had Enums for vehicles, or even bette, structs. But it seems out of scope for this PR